lib/block-supports/spacing.php

```diff
@@ -90,6 +90,57 @@ function gutenberg_skip_spacing_serialization( $block_type ) {
 		$spacing_support['__experimentalSkipSerialization'];
 }
 
+
+/**
+ * Renders the spacing gap support to the block wrapper, to ensure
+ * that the CSS variable is rendered in all environments.
+ *
+ * @param  string $block_content Rendered block content.
+ * @param  array  $block         Block object.
+ * @return string                Filtered block content.
+ */
+function gutenberg_render_spacing_gap_support( $block_content, $block ) {
+	$block_type      = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
+	$has_gap_support = gutenberg_block_has_support( $block_type, array( 'spacing', 'blockGap' ), false );
+	if ( ! $has_gap_support || ! isset( $block['attrs']['style']['spacing']['blockGap'] ) ) {
+		return $block_content;
+	}
+
+	$gap_value = $block['attrs']['style']['spacing']['blockGap'];
+
+	// Skip if gap value contains unsupported characters.
+	// Regex for CSS value borrowed from `safecss_filter_attr`, and used here
+	// because we only want to match against the value, not the CSS attribute.
+	if ( preg_match( '%[\\\(&=}]|/\*%', $gap_value ) ) {
```

Comment:
I wonder if it's the responsibility of this function to do this check (as opposed to presave validation)?

Reply:
Good question: my thinking here was that it'd be good to include some form of CSS value filtering as a defensive conditional so that the rendering logic doesn't depend on the value within the attribute being saved in the correct form, mostly to ensure we don't trust the inputs _too_ much.

The attribute values themselves don't get run through `safecss_filter_attr` when saved (I think!), so this adds a tiny bit of hardening to the render logic, where we assume that we'll have a fairly limited range of desired simple values being entered into the gap block support.

Happy to change this if you think it's better to pull it out or add a wider range of allowed characters.

Reply:
Well I'm not an expert here :) so I'm happy if we keep it this way. Maybe @jorgefilipecosta would have a more informed opinion.